CODE_STYLE.md

```diff
@@ -0,0 +1,3293 @@
+condition](#safe-assignment-in-condition).
+
+* <a name="no-multiline-while-do"></a>
+  Do not use `while/until condition do` for multi-line `while/until`.
+<sup>[[link](#no-multiline-while-do)]</sup>
+
+  ```Ruby
+  # bad
+  while x > 5 do
+    # body omitted
+  end
+
+  until x > 5 do
+    # body omitted
+  end
+
+  # good
+  while x > 5
+    # body omitted
+  end
+
+  until x > 5
+    # body omitted
+  end
+```
+
+* <a name="infinite-loop"></a>
+  Use `Kernel#loop` instead of `while/until` when you need an infinite loop.
+<sup>[[link](#infinite-loop)]</sup>
+
+    ```ruby
+    # bad
+    while true
+      do_something
+    end
+
+    until false
+      do_something
+    end
+
+    # good
+    loop do
+      do_something
+    end
+    ```
+
+* <a name="loop-with-break"></a>
+  Use `Kernel#loop` with `break` rather than `begin/end/until` or
+  `begin/end/while` for post-loop tests.
+<sup>[[link](#loop-with-break)]</sup>
+
+  ```Ruby
+  # bad
+  begin
+    puts val
+    val += 1
+  end while val < 0
+
+  # good
+  loop do
+    puts val
+    val += 1
+    break unless val < 0
+  end
+  ```
+
+* <a name="no-dsl-parens"></a>
+  Omit parentheses around parameters for methods that are part of an internal
+  DSL (e.g. Rake, Rails, RSpec), methods that have "keyword" status in Ruby
+  (e.g. `attr_reader`, `puts`) and attribute access methods. Use parentheses
+  around the arguments of all other method invocations.
+<sup>[[link](#no-dsl-parens)]</sup>
+
+  ```Ruby
+  class Person
+    attr_reader :name, :age
+
+    # omitted
+  end
+
+  temperance = Person.new('Temperance', 30)
+  temperance.name
+
+  puts temperance.age
+
+  x = Math.sin(y)
+  array.delete(e)
+
+  bowling.score.should == 0
+  ```
+
+* <a name="no-braces-opts-hash"></a>
+  Omit the outer braces around an implicit options hash.
+<sup>[[link](#no-braces-opts-hash)]</sup>
+
+  ```Ruby
+  # bad
+  user.set({ name: 'John', age: 45, permissions: { read: true } })
+
+  # good
+  user.set(name: 'John', age: 45, permissions: { read: true })
+  ```
+
+* <a name="no-args-no-parens"></a>
+  Omit parentheses for method calls with no arguments.
+<sup>[[link](#no-args-no-parens)]</sup>
+
+  ```Ruby
+  # bad
+  Kernel.exit!()
+  2.even?()
+  fork()
+  'test'.upcase()
+
+  # good
+  Kernel.exit!
+  2.even?
+  fork
+  'test'.upcase
+  ```
+
+* <a name="single-action-blocks"></a>
+  Use the proc invocation shorthand when the invoked method is the only operation of a block.
+<sup>[[link](#single-action-blocks)]</sup>
+
+  ```Ruby
+  # bad
+  names.map { |name| name.upcase }
+
+  # good
+  names.map(&:upcase)
+  ```
+
+* <a name="single-line-blocks"></a>
+  Prefer `{...}` over `do...end` for single-line blocks.  Avoid using `{...}`
+  for multi-line blocks (multiline chaining is always ugly). Always use
+  `do...end` for "control flow" and "method definitions" (e.g. in Rakefiles and
+  certain DSLs).  Avoid `do...end` when chaining.
+<sup>[[link](#single-line-blocks)]</sup>
+
```

Comment:
As a fan of FP I find this advice poor. There is nothing wrong with multiline chaining, and in that case `{}` works much better.

I think we should allow this. Otherwise, people will cram too much onto one line.